include/cpr/progresscb.h

```diff
@@ -0,0 +1,28 @@
+#ifndef CPR_PROGRESSCB_H
+#define CPR_PROGRESSCB_H
+
+namespace cpr {
+
+class ProgressCallback {
+  #if LIBCURL_VERSION_NUM < 0x072000 // older than 7.32.0 uses the older callback.
+  typedef int(*progress_cb_t)(void*, double, double, double, double);
+  #else
+  typedef int(*progress_cb_t)(void*, int64_t, int64_t, int64_t, int64_t);
```

Comment:
Based on the documentation, the callback requires `curl_off_t` as parameter.
https://curl.haxx.se/libcurl/c/progressfunc.html

Perhaps we should create a `typedef curl_off_t progress_t;` for `curl_off_t` so we do not expose this type.
So other other applications do not need to link against libcurl.

@KingKili, what do you think about this one?

Reply:
Thats true, but we need to make sure that we notice when libcurl changes the type.

Reply:
@ITotalJustice You might want to rebase and have a look at `cprtypes.h` where I defined a wrapper for `curl_off_t` in 943194f16fbb68b406a00e7404321d886031b026.